File: js/problems/1342-number-of-steps-to-reduce-a-number-to-zero.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function (num) {
  if (num === 0) return 0;

  let steps = 0;

  while (num >= 1) {
      num = num % 2 == 0 ? (num / 2) : (num - 1);

      ++steps;
  }

  return steps;
};

File: c/problems/1342-number-of-steps-to-reduce-a-number-to-zero.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
 public:
  int numberOfSteps(int num) {
    if (num == 0) return 0;

    int steps = 0;

    while (num > 0) {
      num = (num % 2 == 0) ? num / 2 : num - 1;

      steps++;
    }

    return steps;
  }
};